contracts/SpokePoolV3Periphery.sol

```diff
@@ -587,6 +587,8 @@ contract SpokePoolV3Periphery is SpokePoolV3PeripheryInterface, Lockable, MultiC
         if (_transferType == TransferType.Approval) _swapToken.forceApprove(_exchange, _swapTokenAmount);
         else if (_transferType == TransferType.Transfer) _swapToken.transfer(_exchange, _swapTokenAmount);
         else {
+            _swapToken.forceApprove(address(permit2), _swapTokenAmount);
```

Comment:
This third TransferType shouldn't be called `EIP1271Signature` i don't think, I think its more appropriately named something to do with Permit2. Technically, we could use another method that requires checking the Periphery's EIP1271Signature that doesn't involve Permit2, right?

Reply:
Yeah that's correct. We could use it for other things. I renamed it to `Permit2Approval` instead here https://github.com/across-protocol/contracts/pull/782/commits/f77331676ed7058639c70429ba370b7db9359112